src/main/java/codeacademy/vigi40/finalexam/chessclub/converters/MemberConverter.java

```diff
@@ -0,0 +1,118 @@
+package codeacademy.vigi40.finalexam.chessclub.converters;
+
+import codeacademy.vigi40.finalexam.chessclub.dto.AddMemberDto;
+import codeacademy.vigi40.finalexam.chessclub.dto.MemberDto;
+import codeacademy.vigi40.finalexam.chessclub.entities.Member;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.format.DateTimeFormatter;
+import java.util.regex.Pattern;
+
+public class MemberConverter {
+
+    public static Member convertAddMemberDtoToEntity(AddMemberDto memberDto) {
+        Member member = null;
+        if (memberDto != null) {
+            if (!validateEmail(memberDto.getEmail())) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Email is not valid");
+            }
+
+            if (!validatePersonalCode(memberDto.getPersonalCode())) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Personal code is not valid");
+            }
+
+            if (!validateStartDate(memberDto.getStartDate())) {
+                throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Start date is not valid");
+            }
+
+            member = new Member();
+            member.setName(memberDto.getName());
+            member.setLastName(memberDto.getLastName());
+            member.setEmail(memberDto.getEmail());
+            member.setPersonalCode(memberDto.getPersonalCode());
+            member.setStartDate(memberDto.getStartDate());
```

Comment:
Sioje vietoje buvo galima prideti validacija, ar gimimo data yra anksciau uz patirties pradzia. Dabar yra galimybe prideti zmogu, kuris gime 2005 metais, o sachmatais zaidzia nuo 1990. Gimimo data validacijai issiskaiciuoti is paduotu asmens kodo.

Reply:
Šita validacija yra 27 eil. `validateStartDate`. Visas validacijas palikau pradžioje metodo, kad nereikėtų naudoti resursų ir atlikti vėlesnio kodo, kai konkrečiai žinome, kodėl jis gali nulūžti

Reply:
Jop, tas tiesa, vieta kurioje aprasei validacija teisinga, tik norejau pamineti, kad truko papildomos validacijos, kuri patikrintu ar gimimo data nera veliau uz data kada pradejo zaisti sachmatais:)